api/src/main/java/net/kyori/adventure/text/format/TextColor.java

```diff
@@ -63,10 +64,46 @@
    * @return a new text colour
    * @since 4.0.0
    */
-  static @NonNull TextColor color(final RGBLike rgb) {
+  static @NonNull TextColor color(final @NonNull RGBLike rgb) {
     return color(rgb.red(), rgb.green(), rgb.blue());
   }
 
+  /**
+   * Creates a new text color, converting the provided HSVLike to the RGB color space.
+   *
+   * @param hsv the hsv value
+   * @return a new text color
+   * @see <a href="https://en.wikipedia.org/wiki/HSL_and_HSV">https://en.wikipedia.org/wiki/HSL_and_HSV</a>
+   * @since 4.6.0
+   */
+  static @NonNull TextColor color(final @NonNull HSVLike hsv) {
+    if(hsv.s() == 0) {
```

Comment:
It might be worth extracting the result of `hsv.s()` etc to local variables? They are used quite a few times in here.